flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/runtime/stream/sql/AggregateITCase.scala

```diff
@@ -1916,4 +1916,44 @@ class AggregateITCase(aggMode: AggMode, miniBatch: MiniBatchMode, backend: State
       )
     assertThat(sink.getRetractResults.sorted).isEqualTo(expected.sorted)
   }
+
+  @TestTemplate
+  def testGroupJsonObjectAggWithRetract(): Unit = {
+    val data = new mutable.MutableList[(Long, String, Long)]
+    data.+=((2L, "Hallo", 2L))
```

Comment:
This can be simplified, e.g., use a for loop

Reply:
Hi lincoln, Thanks for your response. I have added the tests with nested agg and simplify the code.